src/rgw/rgw_auth.h

```diff
@@ -122,14 +122,21 @@ class IdentityApplier : public Identity {
  *  E. execute-commit - commit the modifications from point C. */
 class Completer {
 public:
-  typedef std::unique_ptr<Completer> cmplptr_t;
+  /* It's expected that Completers would tend to implement many interfaces
+   * and be used not only in req_state::auth::completer. Ref counting their
+   * instances woild be helpful. */
+  typedef std::shared_ptr<Completer> cmplptr_t;
```

Comment:
okay, thanks. i was hoping that we could stick with `unique_ptr` for `req_state::completer` and rely on the lifetime of `req_state` to share raw pointers elsewhere, but i see that `RGWRestfulIO` will outlive `req_state`. so using `shared_ptr` is probably easier than allowing filters to be removed from `RGWRestfulIO` before the `req_state` goes away